api/provider/aws/deploy_list.go

```diff
@@ -0,0 +1,72 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (d *DeployProvider) List() ([]models.DeploySummary, error) {
+	taskDefinitionARNs, err := d.listTaskDefinitionARNs()
+	if err != nil {
+		return nil, err
+	}
+
+	summaries, err := d.populateSummariesFromTaskDefinitionARNs(taskDefinitionARNs)
+	if err != nil {
+		return nil, err
+	}
+
+	return summaries, nil
+}
+
+func (d *DeployProvider) listTaskDefinitionARNs() ([]string, error) {
+	taskDefinitionARNs := []string{}
+	fn := func(output *ecs.ListTaskDefinitionsOutput, lastPage bool) bool {
+		for _, taskDefinitionARN := range output.TaskDefinitionArns {
+			taskDefinitionARNs = append(taskDefinitionARNs, aws.StringValue(taskDefinitionARN))
+		}
+		return !lastPage
+	}
+
+	input := &ecs.ListTaskDefinitionsInput{}
+	if err := d.AWS.ECS.ListTaskDefinitionsPages(input, fn); err != nil {
```

Comment:
This implementation is not very efficient. Listing taskDefinitionARNs for a layer0 instance should be done like this for optimum performance: 

* Use [ListTaskDefinitionFamiliesPages](https://github.com/aws/aws-sdk-go/blob/master/service/ecs/api.go#L1944) to get all of the families for a layer0 instance. The `input` should be constructed like so:
```
input := &ecs.ListTaskDefinitionFamiliesInput{}
// familyPrefix is the fully-qualified prefix for your layer0 instance, e.g. `l0-bivers-`
input.SetFamilyPrefix(familyPrefix)
// you probably should double-check this, but I believe we only want to 
// show 'active' task definitions. I think 'inactive' ones have been 'deleted'. 
input.SetStatus(ecs.TaskDefinitionFamilyStatusActive)
```

* Once you have a list of the task definition families, use [ListTaskDefinitionsPages](https://github.com/aws/aws-sdk-go/blob/master/service/ecs/api.go#L6492) for each family returned.  The `input` should be constructed like:
```
input := &ecs.ListTaskDefinitionsInput{}
// note that the 'FamilyPrefix' param is poorly named. This actually needs to be an exact match
// of the family name
input.SetFamilyPrefix(familyName)
// again, you probably should double-check this, but I believe we only want to 
// show 'active' task definitions. I think 'inactive' ones have been 'deleted'. 
input.SetStatus(ecs.TaskDefinitionFamilyStatusActive)
```

Reply:
It's not clear how I would derive the `familyPrefix` if neither the public function List() nor the helper function listTaskDefinitionARNs are passed anything that could help derive what the `familyPrefix` should be.